### ISSUE TITLE
Outdated API?

### ISSUE THREAD

**Issue by pm5** [Leonidas-from-XIV/slacko#43]
With the included `slack_notify` CLI tool one can find a few outdated usage of [Slack Web API](https://api.slack.com/methods):

- Authentication can no longer be done with a `token=...` argument, but must be with `Authorization` header.
- `channels.list` method seems to be replaced by `admin.usergroups.listChannels`.

There also seems to be a lot of methods not supported, such `admins.*`, `reactions.*`, `reminders.*`, `usergroups.*` (replaces `groups.*` probably), etc.

**Comment by Leonidas-from-XIV** [comment]
Yes, indeed, I haven't kept up with the API changes in the last couple of years. It would need some updating to get back to an usable state, but I haven't had the time to do so, given I don't automate Slack much these days anymore.

Pull requests are of course welcome, I'll try to review them quickly.

**Comment by pm5** [comment]
I might be able to help since I'm doing a side project that will use Slack API.

**Comment by Leonidas-from-XIV** [comment]
Sounds good! I've merged your PRs and modernized the library a bit so it is a better place to start.
